core/src/main/java/io/undertow/server/handlers/accesslog/DefaultAccessLogReceiver.java

```diff
@@ -309,11 +325,59 @@ public void rotate() {
         }
     }
 
+    @SuppressWarnings("static-access")
     @Override
     public void close() throws IOException {
-        closed = true;
-        if (stateUpdater.compareAndSet(this, 0, 1)) {
-            logWriteExecutor.execute(this);
+        this.closed = true;
+        if (this.stateUpdater.compareAndSet(this, 0, 3)) {
+            flushAndTerminate();
+            return;
+        } else {
+            // state[1,2] - scheduled or running, attempt schedule hijack
+            if (this.stateUpdater.compareAndSet(this, 1, 3)) {
+                //this means this thread raced against scheduled run(). run() will exist ASAP
```

Comment:
run() will exit

Reply:
yes, as it did not reach any functional block, just might be sleeping, so as soon as it will kick in( either from dormancy or concurrency induced async) it will not enter any IO block.

Reply:
I see

Reply:
run() will exit -> just a typo to be fixed in the comment